Mathlib/Algebra/Group/Subgroup/Finite.lean

```diff
@@ -102,26 +102,24 @@ theorem card_top : Nat.card (⊤ : Subgroup G) = Nat.card G :=
   Nat.card_congr Subgroup.topEquiv.toEquiv
 
 @[to_additive]
-theorem eq_top_of_card_eq [Finite H] (h : Nat.card H = Nat.card G) :
-    H = ⊤ := by
-  have : Nonempty H := ⟨1, one_mem H⟩
-  have h' : Nat.card H ≠ 0 := Nat.card_pos.ne'
-  have : Finite G := (Nat.finite_of_card_ne_zero (h ▸ h'))
-  have : Fintype G := Fintype.ofFinite G
-  have : Fintype H := Fintype.ofFinite H
-  rw [Nat.card_eq_fintype_card, Nat.card_eq_fintype_card] at h
-  rw [SetLike.ext'_iff, coe_top, ← Finset.coe_univ, ← (H : Set G).coe_toFinset, Finset.coe_inj, ←
-    Finset.card_eq_iff_eq_univ, ← h, Set.toFinset_card]
-  congr
+theorem eq_of_le_of_card_ge {H K : Subgroup G} [Finite K] (hle : H ≤ K)
+    (hcard : Nat.card K ≤ Nat.card H) :
+    H = K :=
+  SetLike.coe_injective <| Set.Finite.eq_of_subset_of_card_le (Set.toFinite _) hle hcard
+
+@[to_additive]
+theorem eq_top_of_card_le [Finite G] (h : Nat.card G ≤ Nat.card H) : H = ⊤ :=
+  eq_of_le_of_card_ge le_top (Nat.card_congr (Equiv.Set.univ G) ▸ h)
+
+@[to_additive]
+theorem eq_top_of_card_eq [Finite H] (h : Nat.card H = Nat.card G) : H = ⊤ := by
+  have : Finite G := Nat.finite_of_card_ne_zero (h ▸ Nat.card_pos.ne')
+  exact eq_top_of_card_le _ (Nat.le_of_eq h.symm)
 
 @[to_additive (attr := simp)]
 theorem card_eq_iff_eq_top [Finite H] : Nat.card H = Nat.card G ↔ H = ⊤ :=
   Iff.intro (eq_top_of_card_eq H) (fun h ↦ by simpa only [h] using card_top)
 
-@[to_additive]
-theorem eq_top_of_le_card [Finite G] (h : Nat.card G ≤ Nat.card H) : H = ⊤ :=
```

Comment:
Please leave deprecated aliases if you remove a lemma

Reply:
I shouldn't have changed the name. I've restored the old name.